qlib/SftpPoller.qm

```diff
@@ -499,6 +515,37 @@ public namespace SftpPoller {
             return sftp.stat(path, n_timeout ?? timeout);
         }
 
+        #! check if a remote path is writable
+        /**
+            @throw "REMOTE-DIR-ERROR" this exception is thrown if the remote path does not exist, is not a directory, or is not not writable
+        */
+        checkRemotePath(string path, bool write = False, *timeout n_timeout) {
+            hash h = stat(path, n_timeout);
+            string apath = rootSftpPath + DirSep + path;
```

Comment:
please do not use DirSep for remote files.  DirSep is the directory separator for the local system - which on Windows is '\' - so this will break the path handling on all UNIX SFTP servers at least when the SftpPoller module is used from Windows